docs/code_ref/io.rst

```diff
@@ -85,6 +87,14 @@ file
             -128.03072  , -128.03072  ]], dtype=float32)
    >>> input_asdf.close()  # doctest: +REMOTE_DATA
 
+When saving a Map to ASDF the specific Map source class is not saved, all maps are saved as a `.GenericMap`.
+This comes with some trade-offs.
+If you are using custom map sources defined outside of `sunpy`, and these sources are imported after asdf has been invoked (not imported, but used) for the first time then they will not be registered with the asdf converter.
+Also if the custom map subclass is not present (registered with `sunpy.map.Map`) upon loading of the map, it will be returned as a `.GenericMap`.
+This approach has been chosen despite these limitations so that a map once saved to an ASDF file can be loaded back into a map (even if it is the base `.GenericMap`), rather than the asdf library returning it as a Python dictionary.
```

Comment:
```suggestion
This approach has been chosen despite these limitations so that once a map is saved to an ASDF file it can always be loaded back into a map rather than the asdf library returning it as a Python dictionary.
```